### PR TITLE
KAFKA-14588 ConfigType moved to server-common

### PR DESCRIPTION
**Pull request by nizhikov**
This PR is part of "KAFKA-14588 Move ConfigCommand to tools".
`ConfigCommand` depends on `ConfigType` class.
This PR moves `ConfigType` to "server-commons" module.

Changes are relatively huge so it make sense to review then separately from other.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by nizhikov** [comment]
Hello @mimaison , @jolshan 

Can, you, please, take a look at this changes?
This PR simply move `ConfigType` class from "core" to "server-commons".

**Comment by nizhikov** [comment]
Hello @showuon 

Can, you, please, take a look at this changes?
This PR simply move ConfigType class from "core" to "server-commons".

File: core/src/main/scala/kafka/admin/ConfigCommand.scala
Comment (mimaison) [inline core/src/main/scala/kafka/admin/ConfigCommand.scala]: Why are we using an iterator here? Can we use `val BrokerSupportedConfigTypes: Seq[String] = ConfigType.ALL.asScala :+ BrokerLoggerConfigType :+ ConfigType.CLIENT_METRICS`?

File: core/src/main/scala/kafka/zk/ZkData.scala
Comment (mimaison) [inline core/src/main/scala/kafka/zk/ZkData.scala]: Same as above, can we build this collection without using `JavaConverters`?

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. I left a couple of questions

File: core/src/main/scala/kafka/zk/ZkData.scala
Comment (nizhikov) [reply core/src/main/scala/kafka/zk/ZkData.scala]: Great catch. Thanks. Rewrote like you suggested.

File: core/src/main/scala/kafka/admin/ConfigCommand.scala
Comment (nizhikov) [reply core/src/main/scala/kafka/admin/ConfigCommand.scala]: Great catch. Thanks. Rewrote like you suggested.

**Comment by nizhikov** [comment]
@mimaison 

> This commit cannot be built

Build status is scary, but looks like some tests just failed.
And they not related to my changes.

**Comment by nizhikov** [comment]
@mimaison Are we want to merge this now?

**Comment by mimaison** [comment]
I'm waiting for the CI to finish. I don't want to risk breaking trunk just before logging off for PTO.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Comment by nizhikov** [comment]
@mimaison Thanks you very musch for taking your time to review and merge PR just before holidays :).
Happy New Year!

**Comment by divijvaidya** [comment]
Hey @nizhikov (cc: @mimaison)
I this this PR may have broken a test since the test is failing in trunk 100% of the time after merging this PR. 

See: https://ge.apache.org/scans/tests?search.relativeStartTime=P28D&search.rootProjectNames=kafka&search.tags=trunk&search.timeZoneId=Europe%2FBerlin&tests.container=kafka.admin.ConfigCommandTest&tests.test=shouldNotSupportAlterClientMetricsWithZookeeper() 
Screenshot 2023-12-24 at 20 57 51

The failure message is:
```
org.opentest4j.AssertionFailedError: expected: <client-metrics is not a known entityType. Should be one of List(topics, clients, users, brokers, ips)> but was: <client-metrics is not a known entityType. Should be one of [topics, clients, users, brokers, ips]>
```
which is a result of have different toString() for ConfigType.

Please fix this. I have created a JIRA - https://issues.apache.org/jira/browse/KAFKA-16048

**Comment by nizhikov** [comment]
Will prepare fix in the nearest time.

**Comment by nizhikov** [comment]
Fix is ready #15068

**Comment by divijvaidya** [comment]
Thank you @nizhikov for the quick fix. Merged it.

**Pull request merged**
